Lib/tarfile.py

```diff
@@ -1533,7 +1534,7 @@ def __init__(self, name=None, mode="r", fileobj=None, format=None,
                         self.fileobj.seek(self.offset)
                         break
                     except HeaderError as e:
-                        raise ReadError(str(e))
+                        raise ReadError(str(e)) from None
```

Comment:
Why `from None`? It can remove useful information.

Reply:
The `str(e)` should be preserving the useful bits, while `from None` [alters the traceback](https://bugs.python.org/issue39717?@ok_message=issue%2039717%20versions%20edited%20ok&@template=item#msg362444) to indicate that the error handler is not broken.